include/proxy-wasm/pairs_util.h

```diff
@@ -45,19 +45,20 @@ class PairsUtil {
    * @param size size of the output buffer.
    * @return indicates whether serialization succeeded or not.
    */
-  static bool marshalPairs(const Pairs &pairs, char *buffer, size_t size);
+  static bool marshalPairs(const Pairs &pairs, char *buffer, size_t size, bool is_wasm_byte_order);
```

Comment:
This shouldn't be added this high up the stack. I'm pretty sure that `isWasmByteOrder()` is going to be evaluated and passed by value to `marshalPairs` and other functions, which misses the point adding `is_wasm_byte_order` parameter to macros in order for preprocessors to remove that call when not needed.

Reply:
Moved the calls to isWasmByteOrder() into htowasm/wasmtoh macros, now it should be removed by preprocessor for non-s390x.